src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/ImageResourceIntTest.java

```diff
@@ -65,4 +67,65 @@ public void getImageDataReturnsNotFound() {
 
         assertThatStatus(responseEntity, 404);
     }
+
+    @Test
+    public void putImageReturnsForbiddenForIncorrectRoles() {
+        final var requestEntity = createHttpEntityWithBearerAuthorisationAndBody("ITAG_USER", List.of("WRONG_ROLE"), imageData);
+        final var responseEntity = testRestTemplate.exchange(
+            "/api/images/offender/A1234AA",
+            HttpMethod.PUT,
+            requestEntity,
+            ImageDetail.class
+        );
+
+        assertThatStatus(responseEntity, 403);
+    }
+
+    @Test
+    public void putImageReturnsNotFoundForInvalidOffender() {
+        final var requestEntity = createHttpEntityWithBearerAuthorisationAndBody("ITAG_USER", List.of("ROLE_SYSTEM_USER"), imageData);
+        final var responseEntity = testRestTemplate.exchange(
+            "/api/images/offender/A9999XX",
+            HttpMethod.PUT,
+            requestEntity,
+            ImageDetail.class);
+
+        assertThatStatus(responseEntity, 404);
+    }
+
+    @Test
+    public void putImageUploadsThumbnailByDefault() {
+        final var requestEntity = createHttpEntityWithBearerAuthorisationAndBody("ITAG_USER", List.of("ROLE_SYSTEM_USER"), imageData);
+        final var responseEntity = testRestTemplate.exchange(
+            "/api/images/offender/A1234AI",
+            HttpMethod.PUT,
+            requestEntity,
+            ImageDetail.class);
+
+        assertThatStatus(responseEntity, 200);
+        assertThat(responseEntity.getBody()).isInstanceOf(ImageDetail.class);
+        final var imageDetail = responseEntity.getBody();
+        assertThat(imageDetail).isNotNull();
+        assertThat(imageDetail.getImageId()).isGreaterThan(0);
+        assertThat(imageDetail.getCaptureDate()).isAfter(LocalDate.now().minusDays(1));
+        assertThat(imageDetail.getImageType()).isEqualTo("OFF_BKG");
+    }
+
+    @Test
+    public void putImageUploadsFullSizeImage() {
```

Comment:
Could have a test for a user not in the correct prison

Reply:
I was thinking this was a purely administrative function rather than something to build into a front-end with real prison users. I see why the byte-array type question on the API now, to upload a file directly?

Reply:
But yes,  I can do some test to verify.

Reply:
Actually, I'm not sure how to do this? Integration tests use ITAG_USER, which will have a caseload setup, but are there offenders in the int. test data to whom ITAG_USER does not have access?